### DISCUSSION TITLE
Better interoperability with C# records

### DISCUSSION THREAD

**Discussion by dnperfors** [fsharp/fslang-design#638] [General]
C# 9 included support for records and the `with` keyword to make a copy of the record with some changes. Functionally this is similar to how F# works with records. What I am missing is a way to use F# records in C# using the `with` keyword and a way to use C# records in F#. 

Changing this would (probably) mean that F# records are compiled slightly differently (for example, I *think* the `'<Clone>$'` clone method should be outputted in the IL). Is this something worth investigating and making a proper RFC for?

**Comment by vzarytovskii** [comment]
Yeah, interop with C# records (both ways) is definitely something we should do. RFC would be great. Some things we want to cover are init-only properties, cloning (`with` syntax, probably only one way?), inheritance (C# records support it, F# don't).

**Comment by dsyme** [comment]
@dnperfors Yes, we need to make progress on this. Starting an RFC outlining all the issues would be great

**Reply by dnperfors** [reply]
Will see what I can do the coming days. I am not an expert in F#, but I will see how for I can come.
